slim/datasets/download_and_convert_cifar.py

```diff
@@ -0,0 +1,367 @@
+  'cifar10': 'https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz',
+  'cifar100': 'https://www.cs.toronto.edu/~kriz/cifar-100-python.tar.gz'
+}
+
+_DATA_DIR = {
+  'cifar10': 'cifar-10-batches-py',
+  'cifar100': 'cifar-100-python',
+}
+
+# The number of training files.
+_NUM_TRAIN_FILES = {
+  'cifar10': 5,
+  'cifar100': 1,
+}
+
+# The height and width of each image.
+_IMAGE_SIZE = 32
+
+# The names of the classes.
+_CLASS_NAMES = {
+  'cifar10': [
+    'airplane',
+    'automobile',
+    'bird',
+    'cat',
+    'deer',
+    'dog',
+    'frog',
+    'horse',
+    'ship',
+    'truck',
+  ],
+  'cifar100': [
+  'apple',
+  'aquarium_fish',
+  'baby',
+  'bear',
+  'beaver',
+  'bed',
+  'bee',
+  'beetle',
+  'bicycle',
+  'bottle',
+  'bowl',
+  'boy',
+  'bridge',
+  'bus',
+  'butterfly',
+  'camel',
+  'can',
+  'castle',
+  'caterpillar',
+  'cattle',
+  'chair',
+  'chimpanzee',
+  'clock',
+  'cloud',
+  'cockroach',
+  'couch',
+  'crab',
+  'crocodile',
+  'cup',
+  'dinosaur',
+  'dolphin',
+  'elephant',
+  'flatfish',
+  'forest',
+  'fox',
+  'girl',
+  'hamster',
+  'house',
+  'kangaroo',
+  'keyboard',
+  'lamp',
+  'lawn_mower',
+  'leopard',
+  'lion',
+  'lizard',
+  'lobster',
+  'man',
+  'maple_tree',
+  'motorcycle',
+  'mountain',
+  'mouse',
+  'mushroom',
+  'oak_tree',
+  'orange',
+  'orchid',
+  'otter',
+  'palm_tree',
+  'pear',
+  'pickup_truck',
+  'pine_tree',
+  'plain',
+  'plate',
+  'poppy',
+  'porcupine',
+  'possum',
+  'rabbit',
+  'raccoon',
+  'ray',
+  'road',
+  'rocket',
+  'rose',
+  'sea',
+  'seal',
+  'shark',
+  'shrew',
+  'skunk',
+  'skyscraper',
+  'snail',
+  'snake',
+  'spider',
+  'squirrel',
+  'streetcar',
+  'sunflower',
+  'sweet_pepper',
+  'table',
+  'tank',
+  'telephone',
+  'television',
+  'tiger',
+  'tractor',
+  'train',
+  'trout',
+  'tulip',
+  'turtle',
+  'wardrobe',
+  'whale',
+  'willow_tree',
+  'wolf',
+  'woman',
+  'worm',
+  ]
+}
+
+_COARSE_CLASS_NAMES = [
+  'aquatic_mammals',
+  'fish',
+  'flowers',
+  'food_containers',
+  'fruit_and_vegetables',
+  'household_electrical_devices',
+  'household_furniture',
+  'insects',
+  'large_carnivores',
+  'large_man-made_outdoor_things',
+  'large_natural_outdoor_scenes',
+  'large_omnivores_and_herbivores',
+  'medium_mammals',
+  'non-insect_invertebrates',
+  'people',
+  'reptiles',
+  'small_mammals',
+  'trees',
+  'vehicles_1',
+  'vehicles_2',
+]
+
+_COARSE_LABELS_FILENAME = 'coarse_labels.txt'
+
+def _add_to_tfrecord(filename, tfrecord_writer, dataset, offset=0):
+  """Loads data from the cifar pickle files and writes files to a TFRecord.
+
+  Args:
+    filename: The filename of the cifar pickle file.
+    tfrecord_writer: The TFRecord writer to use for writing.
+    offset: An offset into the absolute number of images previously written.
+
+  Returns:
+    The new offset.
+  """
+  with tf.gfile.Open(filename, 'rb') as f:
+    if six.PY3:
+        data = cPickle.load(f, encoding='bytes')
```

Comment:
Thanks Mark.  I chose the first option.  Tested with python 3, TF 0.11.0rc0.  Works fine.